src/webgpu/util/f32_interval.ts

```diff
@@ -1638,6 +1654,148 @@ export function degreesInterval(n: number): F32Interval {
   return runPointToIntervalOp(toF32Interval(n), DegreesIntervalOp);
 }
 
+/**
+ * Calculate the minor of a NxN matrix.
+ *
+ * The ijth minor of a square matrix, is the N-1xN-1 matrix created by removing
+ * the ith column and jth row from the original matrix.
+ */
+function minorNxN(m: Matrix<number>, col: number, row: number): Matrix<number> {
+  const dim = m.length;
+  assert(m.length === m[0].length, `minorMatrix is only defined for square matrices`);
+  assert(col >= 0 && col < dim, `col ${col} needs be in [0, # of columns '${dim}')`);
+  assert(row >= 0 && row < dim, `row ${row} needs be in [0, # of rows '${dim}')`);
+
+  const result: Matrix<number> = [...Array(dim - 1)].map(_ => [...Array(dim - 1)]);
+
+  const col_indices: number[] = [...Array(dim).keys()].filter(e => e !== col);
+  const row_indices: number[] = [...Array(dim).keys()].filter(e => e !== row);
+
+  col_indices.forEach((c, i) => {
+    row_indices.forEach((r, j) => {
+      result[i][j] = m[c][r];
+    });
+  });
+  return result;
+}
+
+/** Calculate an acceptance interval for determinant(m), where m is a 2x2 matrix */
+function determinant2x2Interval(m: Matrix<number>): F32Interval {
+  assert(
+    m.length === m[0].length && m.length === 2,
+    `determinant2x2Interval called on non-2x2 matrix`
+  );
+  return subtractionInterval(
+    multiplicationInterval(m[0][0], m[1][1]),
+    multiplicationInterval(m[0][1], m[1][0])
+  );
+}
+
+/** Calculate an acceptance interval for determinant(m), where m is a 3x3 matrix */
+function determinant3x3Interval(m: Matrix<number>): F32Interval {
+  assert(
+    m.length === m[0].length && m.length === 3,
+    `determinant3x3Interval called on non-3x3 matrix`
+  );
+
+  // M is a 3x3 matrix
+  // det(M) is A + B + C, where A, B, C are three elements in a row/column times
+  // their own co-factor.
+  // (The co-factor is the determinant of the minor of that position with the
+  // appropriate +/-)
+  // For simplicity sake A, B, C are calculated as the elements of the first
+  // column
+  const A = multiplicationInterval(m[0][0], determinant2x2Interval(minorNxN(m, 0, 0)));
+  const B = multiplicationInterval(
+    m[0][1],
+    multiplicationInterval(-1, determinant2x2Interval(minorNxN(m, 0, 1)))
+  );
+  const C = multiplicationInterval(m[0][2], determinant2x2Interval(minorNxN(m, 0, 2)));
+
+  // Need to calculate permutations, since for fp addition is not transitive,
```

Comment:
nit:  s/transitive/associative/

It's the associativity that's the issue.

  (a+b)+c  is not always a + (b+c).

Cummutativity is   a + b = b + c, and that does hold for float math.